### PR TITLE
Define BDD test steps for extended Type Concept API

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We define the functions corresponding to the BDD test steps introduced for the extended Type Concept API.

## What are the changes implemented in this PR?

New behaviour test steps and minor (mostly cosmetic) fixes for some existing ones.

### CODE REVIEW

File: dependencies/vaticle/artifacts.bzl [hunk 1]
```diff
@@ -29,7 +29,7 @@ def vaticle_typedb_artifact():
         artifact_name = "typedb-server-{platform}-{version}.{ext}",
         tag_source = deployment["artifact.release"],
         commit_source = deployment["artifact.snapshot"],
-        commit = "38773eeffefd63bfa7169a550157add2aca38eb5"
+        tag = "2.9.0",
```
Comment (dmitrii-ubskii) [inline dependencies/vaticle/artifacts.bzl:32]: The typedb-server version where the new protocol methods are implemented.

File: test/behaviour/concept/type/attributetype/AttributeTypeSteps.java [hunk 2]
```diff
@@ -123,34 +123,66 @@ public void attribute_type_as_value_type_does_not_have_any_regex(String typeLabe
     }
 
     @Then("attribute\\( ?{type_label} ?) get key owners contain:")
-    public void attribute_type_get_owners_as_key_contains(String typeLabel, List<String> ownerLabels) {
+    public void attribute_type_get_owners_as_key_contain(String typeLabel, List<String> ownerLabels) {
```
Comment (dmitrii-ubskii) [inline test/behaviour/concept/type/attributetype/AttributeTypeSteps.java:126]: Drive-by: fix the wording of the existing step methods to align better with the BDD step.

File: test/behaviour/concept/type/relationtype/RelationTypeSteps.java [hunk 3]
```diff
@@ -80,16 +80,26 @@ public void relation_type_get_role_type_is_null(String relationLabel, String rol
...
+        assertEquals(isNull, isNull(tx().concepts().getRelationType(relationLabel).asRemote(tx()).getRelatesOverridden(roleLabel)));
+    }
+
     @When("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) set label: {type_label}")
     public void relation_type_get_role_type_set_label(String relationLabel, String roleLabel, String newLabel) {
         tx().concepts().getRelationType(relationLabel).asRemote(tx()).getRelates(roleLabel).asRemote(tx()).setLabel(newLabel);
     }
 
-    @When("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) get label: {type_label}")
+    @Then("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) get label: {type_label}")
```
Comment (dmitrii-ubskii) [inline test/behaviour/concept/type/relationtype/RelationTypeSteps.java:93]: `get` steps should be `Then`-assertions, not `When`-prerequisites.

File: test/behaviour/concept/type/relationtype/RelationTypeSteps.java [hunk 4]
```diff
@@ -111,6 +121,22 @@ public void relation_type_get_related_roles_do_not_contain(String relationLabel,
         for (Label label : roleLabels) assertFalse(actuals.contains(label));
     }
 
+    private Set<Label> relation_type_get_related_explicit_roles_actuals(String relationLabel) {
```
Comment (flyingsilverfin) [inline test/behaviour/concept/type/relationtype/RelationTypeSteps.java:124]: we should normally put the private methods used in other methods below the callers

File: test/behaviour/concept/type/thingtype/ThingTypeSteps.java [hunk 5]
```diff
@@ -362,4 +400,18 @@ public void thing_type_get_playing_roles_do_not_contain(RootLabel rootLabel, Str
...
+        Set<Label> actuals = get_thing_type(rootLabel, typeLabel).asRemote(tx()).getPlaysExplicit().map(Type::getLabel).collect(toSet());
+        assertTrue(actuals.containsAll(roleLabels));
+    }
+
+    @Then("{root_label}\\( ?{type_label} ?) get playing roles explicit do not contain:")
+    public void thing_type_get_playing_roles_explicit_do_not_contain(RootLabel rootLabel, String typeLabel, List<Label> roleLabels) {
+        Set<Label> actuals = get_thing_type(rootLabel, typeLabel).asRemote(tx()).getPlaysExplicit().map(Type::getLabel).collect(toSet());
+        for (Label roleLabel : roleLabels) {
+            assertFalse(actuals.contains(roleLabel));
+        }
```
Comment (flyingsilverfin) [inline test/behaviour/concept/type/thingtype/ThingTypeSteps.java:415]: notice that in some cases you've done `for` loop inline and in some places you've expanded with `{`
we should pick one style and try to be consistent (and particularly within one PR)

**Review by flyingsilverfin — CHANGES_REQUESTED** [review]

File: test/behaviour/concept/type/thingtype/ThingTypeSteps.java (hunk 5 above)
Comment (dmitrii-ubskii) [reply test/behaviour/concept/type/thingtype/ThingTypeSteps.java:417]: Switched to the braced version everywhere since that seems to be the prevailing style in these tests.

File: test/behaviour/concept/type/relationtype/RelationTypeSteps.java (hunk 4 above)
Comment (flyingsilverfin) [reply test/behaviour/concept/type/relationtype/RelationTypeSteps.java:124]: mmm odd style but okay keep it for now

**Review by flyingsilverfin — APPROVED** [review]
